mne_bids/write.py

```diff
@@ -478,13 +477,35 @@ def _scans_tsv(raw, raw_fname, fname, overwrite=False):
           for raw_f in raw_fnames]),
             ('acq_time', [acq_time] * len(raw_fnames))])
 
+    # add source filename if desired
+    if keep_source:
+        data['source'] = [Path(src_fname).name for src_fname in raw.filenames]
+
+        # write out a sidecar JSON if not exists
+        sidecar_json_path = Path(fname).with_suffix('.json')
+        sidecar_json_path = get_bids_path_from_fname(sidecar_json_path)
+        sidecar_json = {'source': 'Original source filename.'}
+
+        if not op.exists(sidecar_json_path):
+            _write_json(sidecar_json_path, sidecar_json)
+        else:
+            update_sidecar_json(sidecar_json_path, sidecar_json)
```

Comment:
Reversing the logic and using `Path`
```suggestion
        if sidecar_json_path.exists():
            update_sidecar_json(sidecar_json_path, sidecar_json)
        else:
            _write_json(sidecar_json_path, sidecar_json)
```

Reply:
The `update_sidecar_json()` line is not covered by tests, do you think you could cook up a tiny test case?

Reply:
apart from that, the sidecar would not really be BIDS, would it? It'd be

`*_scans.json`:

```
{
    "source": "Original source filename."
}
```

but should be something like


{
    "source": {
        "Description": "Original source filename."
    }
}
```

```
    "bci_prediction_valid": {
        "Description": "Did a prediction arrive on time (if not, a random prediction was made) in this trial.",
        "Levels": {
            "1": "True",
            "2": "False"
        }


```

Reply:
Sorry @sappelhoff  not sure what you mean here? It passes the bids validator no?

Reply:
I posted two examples above: The first one is what you currently do in this PR, the second one is valid BIDS - unless I am missing something. :-) 

see: https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#tabular-files

Reply:
Ah I see, okay lmk if the current version is okay then?

Reply:
see https://github.com/mne-tools/mne-bids/pull/890/files#r753073966 :)